commands/filescommands/objectcommands/uploaddir.go

```diff
@@ -0,0 +1,257 @@
+	Action:      actionUploadDir,
+	Flags:       util.CommandFlags(flagsUploadDir, keysUploadDir),
+	BashComplete: func(c *cli.Context) {
+		util.CompleteFlags(util.CommandFlags(flagsUploadDir, keysUploadDir))
+	},
+}
+
+func flagsUploadDir() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "container",
+			Usage: "[required] The name of the container to upload the objects.",
+		},
+		cli.StringFlag{
+			Name:  "dir",
+			Usage: "[optional; required if `stdin` isn't provided] The name the local directory which will be uploaded.",
+		},
+		cli.StringFlag{
+			Name:  "stdin",
+			Usage: "[optional; required if `dir` isn't provided] The field being piped to STDIN. Valid values are: dir.",
+		},
+		cli.StringFlag{
+			Name:  "content-type",
+			Usage: "[optional] The Content-Type header that will be set on all objects.",
+		},
+		cli.IntFlag{
+			Name:  "content-length",
+			Usage: "[optional] The Content-Length header that will be set on all objects.",
+		},
+		cli.StringFlag{
+			Name:  "content-encoding",
+			Usage: "[optional] The Content-Encoding header that will be set on all objects. By default, the uploaded content will be gzipped.",
+		},
+		cli.IntFlag{
+			Name:  "concurrency",
+			Usage: "[optional] The amount of concurrent workers that will upload the directory.",
+		},
+		cli.BoolFlag{
+			Name:  "quiet",
+			Usage: "[optional] By default, every file upload will be outputted. If --quiet is provided, only a final summary will be outputted.",
+		},
+		cli.BoolFlag{
+			Name:  "recurse",
+			Usage: "[optional] By default, only files at the root level of the specified directory are uploaded. If --recurse is provided, the upload will be fully recursive and the entire subtree uploaded.",
+		},
+	}
+}
+
+var keysUploadDir = []string{}
+
+type paramsUploadDir struct {
+	container   string
+	dir         string
+	stream      io.ReadSeeker
+	opts        objects.CreateOpts
+	concurrency int
+	quiet       bool
+	recurse     bool
+}
+
+type commandUploadDir handler.Command
+
+func actionUploadDir(c *cli.Context) {
+	command := &commandUploadDir{
+		Ctx: &handler.Context{
+			CLIContext: c,
+		},
+	}
+	handler.Handle(command)
+}
+
+func (command *commandUploadDir) Context() *handler.Context {
+	return command.Ctx
+}
+
+func (command *commandUploadDir) Keys() []string {
+	return keysUploadDir
+}
+
+func (command *commandUploadDir) ServiceClientType() string {
+	return serviceClientType
+}
+
+func (command *commandUploadDir) HandleFlags(resource *handler.Resource) error {
+	if err := command.Ctx.CheckFlagsSet([]string{"container"}); err != nil {
+		return err
+	}
+
+	c := command.Ctx.CLIContext
+
+	opts := objects.CreateOpts{
+		ContentLength: int64(c.Int("content-length")),
+		ContentType:   c.String("content-type"),
+	}
+
+	if c.IsSet("content-encoding") && c.String("content-encoding") != "gzip" {
+		opts.ContentEncoding = c.String("content-encoding")
+	}
+
+	conc := c.Int("concurrency")
+	if conc <= 0 {
+		conc = 100
```

Comment:
I used this for testing and forgot to change it back. I think it'd be OK for the default, but I'd also be OK with something else.